cloudweatherreport/run.py

```diff
@@ -146,35 +152,108 @@ def load_report(self, datastore, index, test_plan):
                 report.upsert_benchmarks(prev_report.benchmarks)
         return report
 
+    def check_cloud_resource(self, test_plan, cloud_info):
+        """Check if resources are available for a cloud.
+
+        This check against the default resource limits. These limits can be
+        changed by updating the following system environment variables:
+          {CLOUD-NAME}_MACHINE_LIMIT
+          {CLOUD-NAME}_SECURITY_GROUP_LIMIT
+          {CLOUD-NAME}_CPU_LIMIT
+        Example to change the limits for AWS, use AWS_MACHINE_LIMIT,
+        AWS_SECURITY_GROUP_LIMIT, AWS_CPU_LIMIT variables.
+
+        :param test_plan: CWR test plan.
+        :param cloud_info: Cloud info containing region and cloud name.
+        """
+        cloud = cloud_info['cloud-tag'].replace("cloud-", "")
+        region = cloud_info['cloud-region']
+        if (test_plan.cloud_resource is None or
+                test_plan.cloud_resource.get('machines') is None or
+                test_plan.cloud_resource.get('cpus') is None):
+            logging.info(
+                'Skipping cloud resource check. Add "cloud_resource" field in '
+                'the test plan to check for resources before performing tests')
+            return None
+        cloud_env = cloud.upper()
+
+        try:
+            machine_limit = int(
+                os.getenv('{}_MACHINE_LIMIT'.format(cloud_env)))
+        except (TypeError, ValueError):
+            machine_limit = None
+        try:
+            sec_limit = int(
+                os.getenv('{}_SECURITY_GROUP_LIMIT'.format(cloud_env)))
+        except (TypeError, ValueError):
+            sec_limit = None
+        try:
+            cpu_limit = int(os.getenv('{}_CPU_LIMIT'.format(cloud_env)))
+        except (TypeError, ValueError):
+            cpu_limit = None
+
+        try:
+            return is_resource_available(
+                cloud=cloud, region=region,
+                num_of_instances=test_plan.cloud_resource['machines'],
+                num_of_security_groups=1,
+                num_of_cpus=test_plan.cloud_resource['cpus'],
+                instance_limit=machine_limit,
+                security_group_limit=sec_limit,
+                cpu_limit=cpu_limit,
+                credentials_name=os.getenv('{}')
+            )
+        except (CloudNotSupported, UnknownCloudName,
+                UnknownCredentialName) as e:
```

Comment:
I think we should treat `CloudNotSupported` differently; it's not really an error, since you're not pre-screening the list, so we shouldn't log it as one.

I also considered saying that the other errors should be more fatal, but then realized that since this is just a pre-check, it's probably better to fail softly and hope the test can pass anyway.  I do worry a bit about missing errors and not having the pre-checks run consistently.